sql/core/src/main/scala/org/apache/spark/sql/internal/columnNodeSupport.scala

```diff
@@ -74,7 +74,7 @@ private[sql] trait ColumnNodeToExpressionConverter extends (ColumnNode => Expres
           analysis.UnresolvedRegex(columnNameRegex, Some(nameParts), conf.caseSensitiveAnalysis)
 
         case UnresolvedRegex(unparsedIdentifier, planId, _) =>
-          convertUnresolvedAttribute(unparsedIdentifier, planId, isMetadataColumn = false)
+          convertUnresolvedRegex(unparsedIdentifier, planId)
```

Comment:
This seems a bit inconsistent with unresolved attributes. Is this observable?

Reply:
I think it's fine in a way that it's consistent with catalyst ..